iiwa_front:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  # in Hz

  joint_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    state_publish_rate: 1000

  joint_torque_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    gains:
      F_joint_1: {p: 2800.0, i: 80.0, d: 100.0, i_clamp: 320, antiwindup: true}
      F_joint_2: {p: 3500.0, i: 20.0, d: 80.0, i_clamp: 320, antiwindup: true}
      F_joint_3: {p: 2800.0, i: 50.0, d: 40.0, i_clamp: 176, antiwindup: true}
      F_joint_4: {p: 1500.0, i: 10.0, d: 25.0, i_clamp: 176, antiwindup: true}
      F_joint_5: {p: 1800.0, i: 10.0, d: 30.0, i_clamp: 110, antiwindup: true}
      F_joint_6: {p: 1200.0, i: 15.0, d: 15.0, i_clamp: 40, antiwindup: true}
      F_joint_7: {p: 800.0, i: 15.0, d: 15.0, i_clamp: 40, antiwindup: true}
    state_publish_rate: 1000

  joint_position_impedance_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - F_joint_1
      - F_joint_2
      - F_joint_3
      - F_joint_4
      - F_joint_5
      - F_joint_6
      - F_joint_7
    state_publish_rate: 1000

  gazebo_ros_control:
    pid_gains:
      F_joint_1: {p: 700,  i: 0.01, d: 100, i_clamp: 320, antiwindup: true}
      F_joint_2: {p: 10000,  i: 100, d: 100, i_clamp: 320, antiwindup: true}
      F_joint_3: {p: 4500,  i: 35, d: 1, i_clamp: 320, antiwindup: true}
      F_joint_4: {p: 5500,  i: 60, d: 2, i_clamp: 320, antiwindup: true}
      F_joint_5: {p: 1000,  i: 30, d: 0.01, i_clamp: 320, antiwindup: true}
      F_joint_6: {p: 500,  i: 0.1, d: 0.01, i_clamp: 320, antiwindup: true}
      F_joint_7: {p: 100,  i: 0.1, d: 0.01, i_clamp: 320, antiwindup: true}


iiwa_back:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  # in Hz

  joint_position_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    state_publish_rate: 1000

  joint_torque_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    gains:
      B_joint_1: {p: 2800.0, i: 80.0, d: 100.0, i_clamp: 320, antiwindup: true}
      B_joint_2: {p: 3500.0, i: 20.0, d: 80.0, i_clamp: 320, antiwindup: true}
      B_joint_3: {p: 2800.0, i: 50.0, d: 40.0, i_clamp: 176, antiwindup: true}
      B_joint_4: {p: 1500.0, i: 10.0, d: 25.0, i_clamp: 176, antiwindup: true}
      B_joint_5: {p: 1800.0, i: 10.0, d: 30.0, i_clamp: 110, antiwindup: true}
      B_joint_6: {p: 1200.0, i: 15.0, d: 15.0, i_clamp: 40, antiwindup: true}
      B_joint_7: {p: 800.0, i: 15.0, d: 15.0, i_clamp: 40, antiwindup: true}
    state_publish_rate: 1000

  joint_position_impedance_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - B_joint_1
      - B_joint_2
      - B_joint_3
      - B_joint_4
      - B_joint_5
      - B_joint_6
      - B_joint_7
    state_publish_rate: 1000

  gazebo_ros_control:
    pid_gains:
      B_joint_1: { p: 700,  i: 0.01, d: 100, i_clamp: 320, antiwindup: true}
      B_joint_2: { p: 10000,  i: 100, d: 100, i_clamp: 320, antiwindup: true}
      B_joint_3: { p: 4500,  i: 35, d: 1, i_clamp: 320, antiwindup: true}
      B_joint_4: { p: 5500,  i: 60, d: 2, i_clamp: 320, antiwindup: true}
      B_joint_5: { p: 1000,  i: 30, d: 0.01, i_clamp: 320, antiwindup: true}
      B_joint_6: { p: 500,  i: 0.1, d: 0.01, i_clamp: 320, antiwindup: true}
      B_joint_7: { p: 100,  i: 0.1, d: 0.01, i_clamp: 320, antiwindup: true}
